packages/server/src/modules/i18n/index.js

```diff
@@ -0,0 +1,39 @@
+import i18n from 'i18next';
+import * as i18nMiddleware from 'i18next-express-middleware';
+
+import Feature from '../connector';
+import settings from '../../../../../settings';
+import modules from '../../modules';
+
+const addResourcesI18n = () => {
+  for (const localization of modules.localizations) {
+    for (const lang of Object.keys(localization.resources)) {
+      i18n.addResourceBundle(
+        i18n.options.whitelist.filter(lng => lng.indexOf(lang) > -1)[0] || lang,
+        localization.ns,
+        localization.resources[lang]
+      );
+    }
+  }
+};
+
+const initI18n = () => {
+  i18n.use(i18nMiddleware.LanguageDetector).init({
+    fallbackLng: settings.i18n.fallbackLng,
+    resources: {},
+    debug: false, // set true to show logs
+    whitelist: settings.i18n.langList,
+    preload: settings.i18n.langList,
+    detection: {
+      lookupCookie: settings.i18n.cookie
+    }
+  });
+  addResourcesI18n();
+};
+
+export default new Feature({
+  beforeware: app => {
+    initI18n();
```

Comment:
This MUST not be called inside beforeware, as I told before this code must be executed once - on module initialization, not per-request

Reply:
Hi, Victor. The ```beforeware``` function is being called once during the App initialization flow by invoking ```applyBeforeware(app);```. In that function we attach a befoware implementation to an app using ```app.use(something)``` construction. That particular ```something``` is being invoked every request. Please, correct me if I'm wrong.

Reply:
@evgenijolshanskij Yes, I was wrong and you are correct, still I would like to ask to move out `initI18n()` call to the outer level for clarity. E.g.
```
initI18n();

export default new Feature({
  ...
```

Reply:
Well, the thing is that we need modules to be initialized by the time i18n is invoked. Only for that reason we have to do it in that place. On the client side, we encapsulate grabbing localizations to the I18nProvider component that is being used after all the initial flow is completed.